astropy/coordinates/tests/test_separation.py

```diff
@@ -0,0 +1,346 @@
+            246.50823798 * u.deg,
+        ),
+    ],
+    ids=lambda x: x.pytest_id,
+)
+@pytest.mark.parametrize("method", ["position_angle", "separation", "separation_3d"])
+def test_equinox_conversion(coord, other_coord, method):
+    """
+    Regression test for
+        - #868, #891: methods raised errors in case of frames with equinoxes
+        - #3106, #15659: unspecified equinox should be interpreted as frame default,
+            both in `SkyCoord` methods (#3106) and in `BaseCoordinateFrame` methods with
+            `SkyCoord` input (#15659).
+        - #5702: `position_angle()` didn't check if equinoxes differed
+    """
+    assert_quantity_allclose(
+        getattr(coord, method)(other_coord.coord), getattr(other_coord, method)
+    )
+    assert_quantity_allclose(
+        getattr(other_coord.coord, method)(coord),
+        getattr(other_coord, f"reversed_{method}"),
+    )
+
+
+@pytest.mark.parametrize("other_class", [SkyCoord, ICRS])
+@pytest.mark.parametrize("coord_class", [SkyCoord, ICRS])
+def test_separation_3d_dimensionless_distance(coord_class, other_class):
+    assert_quantity_allclose(
+        coord_class(35 * u.deg, 0 * u.deg, 3 * u.one).separation_3d(
+            other_class(125 * u.deg, 0 * u.deg, 4 * u.one)
+        ),
+        5 * u.one,
+    )
+
+
+@pytest.mark.parametrize("dimensionless_class", [SkyCoord, ICRS])
+@pytest.mark.parametrize("length_class", [SkyCoord, ICRS])
+def test_separation_3d_distance_dimension_mismatch(length_class, dimensionless_class):
+    dimensionless_coord = dimensionless_class(1 * u.deg, -2 * u.deg, 14)
+    length_coord = length_class(-1 * u.deg, 2 * u.deg, 21 * u.pc)
+    error_message = (
+        "^Can only apply 'subtract' function to quantities with compatible dimensions$"
+    )
+    with pytest.raises(u.UnitConversionError, match=error_message):
+        dimensionless_coord.separation_3d(length_coord)
+    with pytest.raises(u.UnitConversionError, match=error_message):
+        length_coord.separation_3d(dimensionless_coord)
+
+
+@pytest.mark.parametrize("coord_class", [SkyCoord, ICRS])
+def test_separation_3d_no_distance(coord_class):
+    coord_no_distance = coord_class(0 * u.deg, 0 * u.deg)
+    coord_with_distance = ICRS(0 * u.deg, 0 * u.deg, distance=3 * u.pc)
+    with pytest.raises(
+        ValueError,
+        match="^This object does not have a distance; cannot compute 3d separation.$",
+    ):
+        coord_no_distance.separation_3d(coord_with_distance)
+    with pytest.raises(
+        u.UnitConversionError,
+        match=(
+            "^Can only apply 'subtract' function to quantities with compatible"
+            " dimensions$"
+        ),
+    ):
+        coord_with_distance.separation_3d(coord_no_distance)
+
+
+@pytest.mark.parametrize("coord_class", [SkyCoord, ICRS])
+@pytest.mark.parametrize(
+    "radial_velocity",
+    [
+        pytest.param({}, id="only_one_with_rv"),
+        pytest.param({"radial_velocity": -108 * u.km / u.s}, id="both_have_rv"),
```

Comment:
This seems a bit weird - why check for RV but not proper motion? Maybe parametrize `kwargs` and add that? Though logically, the test cases should then have proper motion too...

Reply:
I had managed to convince myself that we have existing tests that focus on radial velocity, but I can't find any such tests so I must have been mistaken. I agree the parametrization should be expanded.